mpmath/functions/zeta.py

```diff
@@ -288,6 +288,8 @@ def terms():
 
 @defun_wrapped
 def bernpoly(ctx, n, z):
+    # Slow implementation:
+    #return sum(ctx.binomial(n,k)*ctx.bernoulli(k)*z**(n-k) for k in xrange(0,n+1))
```

Comment:
That should be corrected, not just reverted.

Reply:
I don't know how to do that. Can you suggest?

Reply:
As I'd suggested above, probably you should add `(-1)**k` factor.

Reply:
It can't be corrected by simply adding `(-1)**k` as an extra factor inside the summand. **The long code is required** to avoid accuracy issues with (say) `bernpoly(1000,10.5)`.

Reply:
@skirpichev To illustrate what I mean, suppose we have
```python
@defun_wrapped
def bernpoly(ctx, n, z):
    n = int(n)
    return sum([ctx.binomial(n,k) * ctx.bernoulli(k) * (-1)**k * z**(n-k) for k in xrange(0,n+1)])
```
Then `bernpoly` is grossly wrong at large inputs:
```python
>>> from mpmath import *
>>> bernpoly(200, 17.5)
mpf('3.6365715774458614e+244')
>>> bernpoly(1000, 10.5)
mpf('-1.0815008290992428e+1778')
```

Reply:
Indeed, I was misguided by your change in L327.

Reply:
@Parcly-Taxel, in fact, old version for the comment was incorrect as well for same reasons.  But without `(-1)**k` term now it's incorrect also for small arguments, e.g.:
```
>>> n=1;z=mpmath.mpf('1.1'); sum(mpmath.binomial(n,k)*(-1)**k*mpmath.bernoulli(k)*z**(n-k) for k in range(0,n+1))
mpf('0.60000000000000009')
>>> mpmath.bernpoly(n, z)
mpf('0.60000000000000009')
>>> n=1;z=mpmath.mpf('1.1'); sum(mpmath.binomial(n,k)**mpmath.bernoulli(k)*z**(n-k) for k in range(0,n+1))
mpf('2.1000000000000001')
>>> n=4;z=mpmath.mpf('1.3'); sum(mpmath.binomial(n,k)*(-1)**k*mpmath.bernoulli(k)*z**(n-k) for k in range(0,n+1))
mpf('0.11876666666666724')
>>> mpmath.bernpoly(n, z)
mpf('0.11876666666666673')
>>> n=4;z=mpmath.mpf('1.3'); sum(mpmath.binomial(n,k)*mpmath.bernoulli(k)*z**(n-k) for k in range(0,n+1))
mpf('8.9067666666666678')
```

Reply:
Nah, that's just floating-point error. Nothing to change here.

Reply:
The "floating-point error" is about 8.9 in the second example:)

Reply:
If there is any error it's commented out. Do you want me to change the comment?